src/api/python/cvc4.pxi

```diff
@@ -736,18 +736,18 @@ cdef class Solver:
         cdef Term term = Term(self)
         if isinstance(size_or_str, int):
             if val is None:
-                term.cterm = self.csolver.mkBitVector(<int> size_or_str)
+                term.cterm = self.csolver.mkBitVector(<uint32_t> size_or_str)
             else:
-                term.cterm = self.csolver.mkBitVector(<int> size_or_str,
-                                                      <int> val)
+                term.cterm = self.csolver.mkBitVector(<uint32_t> size_or_str,
+                                                      <uint64_t> val)
```

Comment:
Hmm, casting is not a proper fix here since `size_or_str` can be an arbitrary integer (e.g. `2**128`). We should actually convert the integer to a string and pass the value as string to `mkBitVector`. @makaimann what do you think?

Reply:
Don't we also have the same issue with `val`?

Reply:
I agree, a better solution is to convert to a string for the value. The size can't be arbitrary so that can stay the same. Thanks for the PR @NicolaasWeideman! 

You'd need to declare this method: https://github.com/NicolaasWeideman/CVC4/blob/9c208b9cac092adc97a98b49d061715d6ad0e89d/src/api/cvc4cpp.h#L2894

here at this line: https://github.com/NicolaasWeideman/CVC4/blob/9c208b9cac092adc97a98b49d061715d6ad0e89d/src/api/python/cvc4.pxd#L189

Then you should just be able to call `str(val)` and pass it to the function. Let me know if I can help.